commit 30c6977bd7b77108d1b1fcc55400ba74106b64e3
Author: xuty <xty50337@hotmail.com>
Date:   Mon Nov 25 00:19:47 2024 +0800

    Shaft patch: use scrollingDeltaX/Y

diff --git a/Sources/SwiftSDL3/src/video/cocoa/SDL_cocoamouse.m b/Sources/SwiftSDL3/src/video/cocoa/SDL_cocoamouse.m
index 24d7863..95c1cb9 100644
--- a/Sources/SwiftSDL3/src/video/cocoa/SDL_cocoamouse.m
+++ b/Sources/SwiftSDL3/src/video/cocoa/SDL_cocoamouse.m
@@ -506,8 +506,9 @@ void Cocoa_HandleMouseEvent(SDL_VideoDevice *_this, NSEvent *event)
         }
     }
 
-    deltaX = [event deltaX];
-    deltaY = [event deltaY];
+    // Shaft patch: use scrollingDeltaX/Y instead of deltaX/Y for trackpad scrolling
+    deltaX = [event scrollingDeltaX]; 
+    deltaY = [event scrollingDeltaY];
 
     if (seenWarp) {
         deltaX += (lastMoveX - data->lastWarpX);
