diff --git a/Sources/SwiftSDL3/src/events/SDL_mouse.c b/Sources/SwiftSDL3/src/events/SDL_mouse.c
index e3fa3fc..645b86a 100644
--- a/Sources/SwiftSDL3/src/events/SDL_mouse.c
+++ b/Sources/SwiftSDL3/src/events/SDL_mouse.c
@@ -951,10 +951,10 @@ static void SDL_PrivateSendMouseButton(Uint64 timestamp, SDL_Window *window, SDL
         SDL_UpdateMouseFocus(window, mouse->x, mouse->y, buttonstate, true);
     }
 
-    if (buttonstate == source->buttonstate) {
-        // Ignore this event, no state change
-        return;
-    }
+    // if (buttonstate == source->buttonstate) {
+    //     // Ignore this event, no state change
+    //     return;
+    // }
     source->buttonstate = buttonstate;
 
     if (clicks < 0) {

